enhancements/insights/rapid-recommendations.md

```diff
@@ -0,0 +1,548 @@
+
+* The Insights ClusterOperator resource (gathered already). 
+  Red Hat (the Observability Intelligence team responsible for operating the pipeline that processes
+  incoming archives) will monitor the two conditions to detect issues with remote configuration.
+* The config map with the remote configuration used for producing the archive. 
+  This will help reproducing and troubleshooting issues.
+* Metrics about the gathering process.
+  * The total real time of data gathering (gathered already)
+  * The real execution time of each gatherer (gathered already)
+  * The number of records created by each gatherer (gathered already)
+  * Insights Operator CPU usage
+  * Insights Operator memory usage
+
+#### Insights Archive Structure Changes
+
+The new container log data will be saved in a new directory structure similar to that 
+of must-gather archives:
+
+```bash
+<archive_root>/namespaces/<namespace_name>/pods/<pod_name>/<container_name>/current.log
+<archive_root>/namespaces/<namespace_name>/pods/<pod_name>/<container_name>/previous.log
+```
+
+Each file will preserve the original (chronological) line order.
+
+Container logs gathered by existing gatherers will remain in their current locations for now. 
+We will seek unification in future updates. This includes data gathered by:
+
+* [ClusterOperatorPodsAndEvents](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#clusteroperatorpodsandevents)
+* [ContainersLogs](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#containerslogs) (conditional gathering)
+* [KubeControllerManagerLogs](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#kubecontrollermanagerlogs)
+* [LogsOfNamespace](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#logsofnamespace) (conditional gathering)
+* [OpenShiftAPIServerOperatorLogs](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#openshiftapiserveroperatorlogs)
+* [OpenshiftAuthenticationLogs](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#openshiftauthenticationlogs)
+* [OpenshiftSDNControllerLogs](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#openshiftsdncontrollerlogs)
+* [OpenshiftSDNLogs](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#openshiftsdnlogs)
+* [SAPVsystemIptablesLogs](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#sapvsystemiptableslogs)
+* [SchedulerLogs](https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md#schedulerlogs)
+
+
+### Risks and Mitigations
+
+#### Remote Configuration Service Inaccessible
+
+Risks:
+
+* The configured remote configuration service endpoint is invalid.
+* The remote configuration is not available temporarily (service down, network configuration issues).
+
+Mitigations:
```

Comment:
Can we bake in a best-effort fallback?  It doesn't have to be great, but it seems as though something would be better than nothing.

Reply:
Just to add a little more context, we are looking to solve two things with this:

1. David's request
2. A flaw in the original design where the Insights Operator could try to use an incompatible configuration after upgrade when the remote configuration service wouldn't be available.

To solve it, we want to "reset" the configuration during cluster/ Insights Operator upgrade to a one "hardcoded" in the new Insights Operator version. Connected clusters would replace the hardcoded configuration with a fresh configuration from the remote configuration service before first use. If the remote configuration service were unavailable and for disconnected/air-gapped clusters, the Insights Operator would still have a compatible configuration to work with.

The "hardcoded" configuration would be updated in every OCP release (including z-stream releases).